lib/graphql/tracing/data_dog_tracing.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module GraphQL
+  module Tracing
+    class DataDogTracing < PlatformTracing
+      self.platform_keys = {
+        'lex' => 'lex.graphql',
+        'parse' => 'parse.graphql',
+        'validate' => 'validate.graphql',
+        'analyze_query' => 'analyze.graphql',
+        'analyze_multiplex' => 'analyze.graphql',
+        'execute_multiplex' => 'execute.graphql',
+        'execute_query' => 'execute.graphql',
+        'execute_query_lazy' => 'execute.graphql',
+      }
+
+      def platform_trace(platform_key, key, data)
+        blacklisted_variables = options.fetch(:blacklisted_variables, [])
+        service = options.fetch(:service, 'ruby-graphql')
+
+        pin = Datadog::Pin.get_from(self)
+        unless pin
+          pin = Datadog::Pin.new(service)
+          pin.onto(self)
+        end
+
+        pin.tracer.trace(platform_key, service: pin.service) do |span|
+          if key == 'execute_multiplex'
+            span.resource = data[:multiplex].queries.map(&:selected_operation_name).join(', ')
+          end
+
+          if key == 'execute_query'
+            span.set_tag(:selected_operation_name, data[:query].selected_operation_name)
+            span.set_tag(:selected_operation_type, data[:query].selected_operation.operation_type)
+            span.set_tag(:query_string, data[:query].query_string)
+            span.set_tag(:variables, scrub_variables(data[:query].variables.to_h), blacklisted_variables)
+          end
+          yield
+        end
+      end
+
+      def platform_field_key(type, field)
+        "#{type.name}.#{field.name}"
+      end
+
+      def scrub_variables(variables, blacklisted_variables)
+        Scrubber::Variables.perform(variables, blacklisted_variables)
```

Comment:
This scrubbing seems like a good idea!

I'm still a bit afraid of a blacklist approach versus a whitelist approach, though. For queries I think it's fine, but for mutations less so. How would you feel about an option that removes arguments from mutations, and users can toggle this when setting the tracing? So, something along the lines of:

```ruby
use DataDogTracing, send_mutation_variables: true
```